scenariolib/visit.go

```diff
@@ -189,15 +189,17 @@ func (v *Visit) sendViewEvent(rank int, contentType string, pageViewField string
 
 	event.PageURI = v.LastResponse.Results[rank].ClickURI
 	event.PageTitle = v.LastResponse.Results[rank].Title
-	// event.ContentType = contentType
-	// event.ContentIDKey = "@" + pageViewField
+	event.ContentType = contentType
+	event.ContentIDKey = "@" + pageViewField
 	event.PageReferrer = v.Referrer
 
-	// if contentIDValue, ok := v.LastResponse.Results[rank].Raw[pageViewField].(string); ok {
-	// 	event.ContentIDValue = contentIDValue
-	// } else {
-	// 	return fmt.Errorf("Cannot convert %s field %s value to string", v.LastResponse.Results[rank].Raw[pageViewField], pageViewField)
-	// }
+	v.decorateCustomMetadata(event.ActionEvent, customData)
+
+	if contentIDValue, ok := v.LastResponse.Results[rank].Raw[pageViewField].(string); ok {
+		event.ContentIDValue = contentIDValue
+	} else {
+		return fmt.Errorf("Cannot convert %s field %s value to string", v.LastResponse.Results[rank].Raw[pageViewField], pageViewField)
```

Comment:
Can the error happen for other reasons? Like when `Results[rank]` is null? or `Raw[pageViewField]` is not defined?

Reply:
It cannot be executed if the LastResponse is nil or there were no results. 
Cannot be executed if you specify a clickRank higher than the number of results returned by the search.

But yes if the field does not exist on the result it will return that error.

I added a better error handling.